test/testSetup.ts

```diff
@@ -99,11 +103,16 @@ export class TestSetup {
     await expect(uncaughtErrors.length).toBe(0);
   }
 
-  public async setUpTestingEnvironment(): Promise<void> {
+  public async setUpTestingEnvironment(repo?: constants.repoKeywords): Promise<void> {
     utilities.log('');
     utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
 
-    this.tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
+    if (repo) {
+      const repoPath = path.join(__dirname, '..', repo);
+      await gitClone(constants.projectMaps.get(repo)!, repoPath); // clone the project
+      Env.getInstance().useExistingProject = repoPath; // set the existing project file path, value will be overriden even if USE_EXISTING_PROJECT_PATH exists
```

Comment:
Modifying this var will have a negative effect on other tests that might run along with the one that is using a specific repository.

The value set wuen initilizing env var should not be modified to suit the needs of this change.

While establishing the project that will be used
- The test author wants a new project
- The test author wants a project cloned into the test
- The test author will use the project specified by Env.getInstance().useExistingProject
- The configuration of one test shall not affect the state of other tests